### PR TITLE
chore: Private Embed Code split for 1-click upgrade downgrade

### PR DESCRIPTION
**Pull request by ankitsrivas14**
Code splitting for Private Embed feature
>
>
#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith-ee/issues/1685

>
#### Type of change
- New feature (non-breaking change which adds functionality)
- Chore (housekeeping or task changes that don't impact user perception)
>
>
>
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [x] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by hetunandu — APPROVED** [review]

File: app/client/src/ce/selectors/rampSelectors.tsx [hunk 1]
```diff
@@ -35,6 +36,8 @@ export const showProductRamps = (
   rampName: string,
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   isEnterpriseOnlyFeature = false,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  featureFlagName?: FeatureFlag,
```
Comment (ankitakinger) [inline app/client/src/ce/selectors/rampSelectors.tsx:40]: are we using this newly added parameter inside?

File: app/client/src/pages/Editor/AppSettingsPane/AppSettings/EmbedSettings/index.tsx [hunk 2]
```diff
@@ -20,8 +20,12 @@ import {
   PERMISSION_TYPE,
 } from "@appsmith/utils/permissionHelpers";
 import MakeApplicationForkable from "./MakeApplicationForkable";
-import EmbedSnippetTab from "@appsmith/pages/Applications/EmbedSnippetTab";
+// eslint-disable-next-line
+import CE_EmbedSnippetTab from "ce/pages/Applications/EmbedSnippetTab";
+import EE_EmbedSnippetTab from "ee/pages/Applications/EmbedSnippetTab";
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/AppSettingsPane/AppSettings/EmbedSettings/index.tsx:25]:
```suggestion
import EE_EmbedSnippetTab from "@appsmith/pages/Applications/EmbedSnippetTab";
```

File: app/client/src/pages/Editor/EditorHeader.tsx [hunk 3]
```diff
@@ -100,7 +100,9 @@ import EndTour from "./GuidedTour/EndTour";
 import { GUIDED_TOUR_STEPS } from "./GuidedTour/constants";
 import { viewerURL } from "RouteBuilder";
 import { useHref } from "./utils";
-import EmbedSnippetForm from "@appsmith/pages/Applications/EmbedSnippetTab";
+// eslint-disable-next-line
+import CE_EmbedSnippetForm from "ce/pages/Applications/EmbedSnippetTab";
+import EE_EmbedSnippetForm from "ee/pages/Applications/EmbedSnippetTab";
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/EditorHeader.tsx:105]:
```suggestion
import EE_EmbedSnippetForm from "@appsmith/pages/Applications/EmbedSnippetTab";
```

File: app/client/src/ce/selectors/rampSelectors.tsx (hunk 1 above)
Comment (ankitsrivas14) [reply app/client/src/ce/selectors/rampSelectors.tsx]: No, we are not using it in the CE version but using in the EE.
Usage is the same as other parameter `isEnterpriseOnlyFeature`

File: app/client/src/pages/Editor/AppSettingsPane/AppSettings/EmbedSettings/index.tsx (hunk 2 above)
Comment (ankitsrivas14) [reply app/client/src/pages/Editor/AppSettingsPane/AppSettings/EmbedSettings/index.tsx]: Updated

File: app/client/src/pages/Editor/EditorHeader.tsx (hunk 3 above)
Comment (ankitsrivas14) [reply app/client/src/pages/Editor/EditorHeader.tsx]: Updated

**Review by ankitakinger — DISMISSED** [review]

**Review by ankitakinger — DISMISSED** [review]

**Review by dipyamanbiswas07 — DISMISSED** [review]

**Review by dipyamanbiswas07 — APPROVED** [review]
LGTM

**Review by ankitakinger — APPROVED** [review]

**Pull request merged**